distribution/integration/tests-integration/user-core/src/test/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManagerTest.java

```diff
@@ -0,0 +1,285 @@
+package org.wso2.carbon.user.core.ldap;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.h2.jdbcx.JdbcDataSource;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.testng.Assert;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+import org.wso2.carbon.caching.impl.CachingConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.claim.ClaimManager;
+import org.wso2.carbon.user.core.claim.ClaimMapping;
+
+import java.io.File;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+import javax.cache.Cache;
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanServer;
+import javax.management.MBeanServerFactory;
+import javax.management.MalformedObjectNameException;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.BasicAttribute;
+import javax.naming.directory.BasicAttributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+
+/**
+ * Integration test for ReadOnlyLDAPUserStoreManager.
+ * Uses embedded LDAP to check the functionality of the class.
+ *
+ */
+public class ReadOnlyLDAPUserStoreManagerTest {
+
+    private static final Log log = LogFactory.getLog(ReadOnlyLDAPUserStoreManagerTest.class);
+
+    private static final String MOBILE_CLAIM = "www.wso2.org/mobile";
```

Comment:
Should claims start with http://. for ex. http://wso2.org/mobile? Or is this a custom claim defined by you?

Reply:
This does not need to be any specific claim, and is arbitrary. Claim mapping is done as mock.